main.go

```diff
@@ -20,503 +20,16 @@
-	monitor.EnableStackdriverExporter(flags.StackdriverExportInterval)
-	monitor.EnableOpenTelemetryCollectorExporter(flags.OtelCollectorAddress)
-
-	// Mount, writing information about our progress to the writer that package
-	// daemonize gives us and telling it about the outcome.
-	var mfs *fuse.MountedFileSystem
-	{
-		mfs, err = mountWithArgs(bucketName, mountPoint, flags, mountConfig)
-
-		// This utility is to absorb the error
-		// returned by daemonize.SignalOutcome calls by simply
-		// logging them as error logs.
-		callDaemonizeSignalOutcome := func(err error) {
-			if err2 := daemonize.SignalOutcome(err); err2 != nil {
-				logger.Errorf("Failed to signal error to parent-process from daemon: %v", err2)
-			}
-		}
-
-		markSuccessfulMount := func() {
-			// Print the success message in the log-file/stdout depending on what the logger is set to.
-			logger.Info(SuccessfulMountMessage)
-			callDaemonizeSignalOutcome(nil)
-		}
-
-		markMountFailure := func(err error) {
-			// Printing via mountStatus will have duplicate logs on the console while
-			// mounting gcsfuse in foreground mode. But this is important to avoid
-			// losing error logs when run in the background mode.
-			logger.Errorf("%s: %v\n", UnsuccessfulMountMessagePrefix, err)
-			err = fmt.Errorf("%s: mountWithArgs: %w", UnsuccessfulMountMessagePrefix, err)
-			callDaemonizeSignalOutcome(err)
-		}
-
-		if err != nil {
-			markMountFailure(err)
-			return err
-		}
-		if !isDynamicMount(bucketName) {
-			switch flags.MetadataPrefetchOnMount {
-			case config.MetadataPrefetchOnMountSynchronous:
-				if err = callListRecursive(mountPoint); err != nil {
-					markMountFailure(err)
-					return err
-				}
-			case config.MetadataPrefetchOnMountAsynchronous:
-				go func() {
-					if err := callListRecursive(mountPoint); err != nil {
-						logger.Errorf("Metadata-prefetch failed: %v", err)
-					}
-				}()
-			}
-		}
-		markSuccessfulMount()
-	}
-
-	// Let the user unmount with Ctrl-C (SIGINT).
-	registerSIGINTHandler(mfs.Dir())
-
-	// Wait for the file system to be unmounted.
-	err = mfs.Join(context.Background())
-
-	monitor.CloseStackdriverExporter()
-	monitor.CloseOpenTelemetryCollectorExporter()
-
-	if err != nil {
-		err = fmt.Errorf("MountedFileSystem.Join: %w", err)
-		return
-	}
-
-	return
-}
-
-func run() (err error) {
-	// Set up the app.
-	app := newApp()
-
-	var appErr error
-	app.Action = func(c *cli.Context) {
-		appErr = runCLIApp(c)
-	}
-
-	// Run it.
-	err = app.Run(os.Args)
-	if err != nil {
-		return
-	}
-
-	err = appErr
-	return
-}
-
-// Handle panic if the crash occurs during mounting.
-func handlePanicWhileMounting() {
-	// Detect if panic happens in main go routine.
-	a := recover()
-	if a != nil {
-		logger.Fatal("Panic: %v", a)
-	}
-}
-
 func main() {
 	if strings.ToLower(os.Getenv("ENABLE_GCSFUSE_VIPER_CONFIG")) == "true" {
 		cmd.Execute()
 		return
 	}
-	defer handlePanicWhileMounting()
```

Comment:
Nit: Shall we make this first line of main and remove it from ExecuteLegacyMain for readability?

Reply:
In Cobra, the main is usually empty and the logic is implemented in the corresponding commands in cmd/ package. There is a good reason for it. Consider that GCSFuse is an application that can potentially perform several functions:
* Mounting
* Displaying the version
* Generating a config (hypothetical)

setting up profiling is something that is probably associated with mounting and would not be needed by the other features. So, it ought to be implemented in the Cobra command for mount and not in main. WDYT?

Reply:
I see, that makes sense. Guess we need to refactor legacy main in future in that case (we can create a backlog ticket for this).
However, I still think `defer handlePanic()` should still be part of main. WDYT?

Reply:
I'm still not sure if we should move handlePanicWhileMounting() to main since it seems useful only for mount command. Do you think, a command such as `gcsfuse version` requires to recover?

Reply:
Nvm, thanks for the explanation. I see that handlePanicWhileMounting() just logs the panic. I've renamed the method to indicate that.